adjust

```diff
@@ -381,7 +381,11 @@ def raw_query(appname, desc, pod_debug=False):
         # list of pods, for runtime_id
         try:
             rs = k_get(appname, ["-l", sel, "rs"])['items']
-            latest_rs = max(rs, key=lambda r: int(r['metadata']['annotations']['deployment.kubernetes.io/revision']))
+            dep_rs = [x for x in rs if any(y['uid'] == dep.get('metadata', {}).get('uid') for y in x.get('metadata', {}).get('ownerReferences', []))]
```

Comment:
1) this functionality seems to appear in a couple places, if that is the case it may be worth extracting in a separate function, i.e. `get_latest_rs_for_deployment()` 

2) I am not sure if it is possible to have a k8s object without 'deployment.kubernetes.io/revision' annotation, but to be on the safe side and consistent with the test of the code, we should probably use dict.get('key') instead of dict['key'] for the full path, including metadata.